src/main/scala/abt/Abt.scala

```diff
@@ -1,15 +1,20 @@
 package abt
 
-import scalaz._, Scalaz._
+import scalaz._
 
-/** @tparam V Type of variables
+/** @tparam S Sort (syntactic category)
+  * @tparam V Type of variables
   * @tparam O underlying AST
   * @tparam T Abt concrete instance
   */
 trait Abt[S, V, O, T] {
-  import View._
-  def check[M](view: View[V, O, T], valence: Valence[S])(implicit M: MonadVar[V, ?], S: Equal[S]): EitherT[M, AbtError, T]
-  def infer[M: MonadVar[V, ?]](t: T): M[(Valence[S], View[V, O, T])]
+  def check[M[_, _]](view: View[V, O, T], valence: Valence[S])
+                    (implicit ME: MonadError[M, AbtError],
+                              MV: MonadVar[M[AbtError, ?], V],
+                              S: Equal[S])
+                    : M[AbtError, T]
```

Comment:
This is odd, right? We introduced these changes during the floobits session, right? Why weren’t they part of my original commit?